File: c/47.c
/* 134043 */
#include <stdio.h>
#include <stdlib.h>
#include "euler.h"

#define N 4
#define M 200000

int main(int argc, char **argv)
{
    int n, i, found;
    char *sieve = nfactor_sieve(M);
    for (n = 646; n < M - N; n++)
    {
        found = 1;
        for (i = 0; i < N; i++)
        {
            if (sieve[n+i] != N)
            {
                found = 0;
                break;
            }
        }
        if (found)
        {
            printf("%d\n", n);
            break;
        }
    }
    free(sieve);
    return 0;
}

File: py/47.py
# 134043
import euler
N = 4
M = 200000
tgt = [N for i in xrange(N)]
factors = euler.nfactor_sieve(M)
for n in xrange(M-N):
    if factors[n:n+N] == tgt:
        print n
        break
